demos/classification_benchmark_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,549 @@
+    std::vector<cv::gapi::wip::onevpl::CfgParam> source_cfgs;
+    std::stringstream params_list(cfg_params);
+    std::string line;
+    while (std::getline(params_list, line, ',')) {
+        source_cfgs.push_back(createFromString(line));
+    }
+    return source_cfgs;
+}
+
+static cv::gapi::GKernelPackage getKernelPackage(const std::string& type) {
+    if (type == "opencv") {
+        return cv::gapi::combine(cv::gapi::core::cpu::kernels(), cv::gapi::imgproc::cpu::kernels());
+    } else if (type == "fluid") {
+        return cv::gapi::combine(cv::gapi::core::fluid::kernels(), cv::gapi::imgproc::fluid::kernels());
+    } else {
+        throw std::logic_error("Unsupported kernel package type: " + type);
+    }
+    GAPI_Assert(false && "Unreachable code!");
+}
+}  // namespace util
+
+namespace nets {
+G_API_NET(Classification, <cv::GMat(cv::GMat)>, "classification");
+}
+
+struct IndexScore {
+    using IndicesStorage = std::list<size_t>;
+    using ConfidenceMap = std::multimap<float, typename IndicesStorage::iterator>;
+
+    using ClassDescription = std::tuple<float, size_t, std::string>;
+    using LabelsStorage = std::vector<ClassDescription>;
+
+    IndexScore() = default;
+
+    void getScoredLabels(const std::vector<std::string> &in_labes,
+                         LabelsStorage &out_scored_labels_to_append) const {
+        try {
+            // fill starting from max confidence
+            for (auto conf_index_it = max_confidence_with_indices.rbegin();
+                 conf_index_it != max_confidence_with_indices.rend();
+                 ++conf_index_it) {
+                std::string str_label = in_labes.at(*conf_index_it->second);
+                out_scored_labels_to_append.emplace_back(conf_index_it->first, *conf_index_it->second,  std::move(str_label));
+            }
+        } catch (const std::out_of_range& ex) {
+            throw std::out_of_range(std::string("Provided labels file doesn't contain detected class index\nException: ") + ex.what());
+        }
+    }
+
+    static IndexScore create_from_array(const float *out_blob_data_ptr, size_t out_blob_element_count,
+                                        size_t top_k_amount) {
+        IndexScore ret;
+        if (!out_blob_data_ptr) {
+            return IndexScore();
+        }
+        // find top K
+        size_t i = 0;
+        // fill & sort topK with first K elements of N array: O(K*Log(K))
+        while(i < std::min(top_k_amount, out_blob_element_count)) {
+            ret.max_element_indices.push_back(i);
+            ret.max_confidence_with_indices.emplace(out_blob_data_ptr[i], std::prev(ret.max_element_indices.end()));
+            i++;
+        }
+
+        // search K elements through remnant N-K array elements
+        // greater than the minimum element in the pivot topK
+        // O((N-K)*Log(K))
+        for (i = top_k_amount; i < out_blob_element_count && !ret.max_confidence_with_indices.empty(); i++) {
```

Comment:
1. If it's 0, why would someone call it in first place? I believe you should handle this as an error or ignore.
2. Agree
3. Will be discussed in https://github.com/openvinotoolkit/open_model_zoo/pull/3800#discussion_r1267996567
4. Let's hold this conversation until 3. is resolved, because it is essentially the same to me

Reply:
removed i assignment, and empty check. Added contract check in the beginning